datadog_checks_base/datadog_checks/base/checks/base.py

```diff
@@ -746,10 +746,11 @@ def event(self, event):
         :param ev event: the event to be sent.
         """
         # Enforce types of some fields, considerably facilitates handling in go bindings downstream
-        for key in event:
-            # Ensure strings have the correct type
+        for key, value in list(iteritems(event)):
+            if not isinstance(value, binary_type):
+                continue
             try:
-                event[key] = to_string(event[key])
+                event[key] = ensure_unicode(event[key])
```

Comment:
This seems to lead to a behaviour change. Before native string is assigned to `event[key]`, but now, it's always `unicode`.

Reply:
Yes, it was a mistake, changed to stick to `to_string` and check for `text_type`. :+1: